compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -192,7 +192,19 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                             );
                             if self.fn_self_span_reported.insert(fn_span) {
                                 err.span_note(
-                                    self_arg.span,
+                                    // Check whether the source is accessible
+                                    if self
+                                        .infcx
+                                        .tcx
+                                        .sess
+                                        .source_map()
+                                        .span_to_snippet(self_arg.span)
+                                        .is_ok()
+                                    {
```

Comment:
It would be nice if we could add something akin to `span_note_with_alternative` so that we could provide a fallback span.